hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -72,13 +72,13 @@ spec = do
             reqSn = NetworkEvent $ ReqSn alice 1 []
             snapshot1 = Snapshot 1 mempty []
             ackFrom sk vk = NetworkEvent $ AckSn vk (sign sk snapshot1) 1
-        s1 <- assertNewState $ update env ledger s0 reqSn
-        s2 <- assertNewState $ update env ledger s1 (ackFrom carolSk carol)
-        s3 <- assertNewState $ update env ledger s2 (ackFrom aliceSk alice)
+        s1 <- assertNewState s0 $ update env ledger s0 reqSn
+        s2 <- assertNewState s1 $ update env ledger s1 (ackFrom carolSk carol)
+        s3 <- assertNewState s2 $ update env ledger s2 (ackFrom aliceSk alice)
```

Comment:
Maybe we should be using the `StateT HeadState` here. We are only interested in the final state, so we could be do something like 

```suggestion
        s3 <- flip evalStateT s0 $
            applyEvent (update env ledger) reqSn >>=
            applyEvent (update env ledger) (ackFrom carolSk carol) >>=
            applyEvent (update env ledger) (ackFrom aliceSk alice)
```

Reply:
Thought about it but decided to leave it as it is so the test is consistent with the rest 
If you consider it fine, ofc I will move forward on this!

Reply:
If it works out nicely, you can apply this technique to all tests. Whatever works better in your opinion.

Reply:
I tried this out and the problem I'm having is that applyEvent results in () so I am not being able to add assertions on the outcome state, as we used to.
wdyt?